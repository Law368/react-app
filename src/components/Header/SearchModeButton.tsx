import React, {MouseEventHandler} from 'react';
import log from '../../helpers/log';

export default function SearchModeButton(props: {
    name: string;
    isActive: boolean;
    handleClick: Function;
}) {
    return (
        <button
            type="button"
            className={`search__modeButton ${
                props.isActive ? 'search__modeButton--active' : ''
            }`}
            onClick={props.handleClick(props.name)}
        >
            {props.name}
        </button>
    );
}
