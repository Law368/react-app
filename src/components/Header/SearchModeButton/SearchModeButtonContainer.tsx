import React from 'react';
import {useDispatch, useSelector} from 'react-redux';
import {searchMode} from '../../../actionCreators/searchMode';
import {Search} from '../../../enums/enum';
import {State} from '../../../reducers/commonReducer';
import SearchModeButton from './SearchModeButton';

export function SearchModeButtonContainer() {
    const searchType = useSelector((state: State) => state.searchMode);
    const dispatch = useDispatch();

    const handleActiveClass = (name: string) => () => {
        if (name === 'title') {
            dispatch(searchMode(Search.Title));
        }
        if (name === 'genre') {
            dispatch(searchMode(Search.Genre));
        }
    };

    // TODO: переделать handler на 2 функции для title и для genre
    return (
        <>
            <SearchModeButton
                name={'title'}
                isActive={searchType === Search.Title}
                handleClick={handleActiveClass}
            />
            <SearchModeButton
                name={'genre'}
                isActive={searchType === Search.Genre}
                handleClick={handleActiveClass}
            />
        </>
    );
}

// 1. Правильно ли составлено переключение активного класса? Не нужно ли менять класс каким то иным способом, не через event target ?
// Получать из store значение типа поиска и в зависимости от него менять имя класса
// 2. Какой тип должен быть у event параметра функции, чтобы у него был доступ к currentTarget и preventDefault() ?
// 3. На настоящий момент код дублируется в 2 компонентах. Если вынести это в отдельную функцию, то как это должно быть структурировано в папках проекта ?

// TODO: Добавить выбранный фильм в store свойство CurrentMovie со всеми его значениями.
// Добавляю данные из стора в хедер после клика по плакату
// производится поиск по первому жанру из свойств фильма - get запрос movie.genre[0]
