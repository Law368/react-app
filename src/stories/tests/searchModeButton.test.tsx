import React from 'react';
import renderer from 'react-test-renderer';

import SearchModeButton from '../../components/Header/SearchModeButton/SearchModeButton';

it('Search mode button should be active', () => {
    const element = renderer
        .create(
            <SearchModeButton
                name={'Test Button'}
                isActive={true}
                handleClick={() => {}}
            />
        )
        .toJSON();
    expect(element).toMatchSnapshot();
});

it('Search mode button should not be active', () => {
    const element = renderer
        .create(
            <SearchModeButton
                name={'Test Button'}
                isActive={false}
                handleClick={() => {}}
            />
        )
        .toJSON();
    expect(element).toMatchSnapshot();
});
