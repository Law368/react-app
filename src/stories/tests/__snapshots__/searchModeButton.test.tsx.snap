// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search mode button should be active 1`] = `
<button
  className="search__modeButton search__modeButton--active"
  type="button"
>
  Test Button
</button>
`;

exports[`Search mode button should not be active 1`] = `
<button
  className="search__modeButton "
  type="button"
>
  Test Button
</button>
`;

exports[`Should be active 1`] = `
<button
  className="search__modeButton "
  type="button"
>
  Test Button
</button>
`;
